qcodes/tests/dataset/test_descriptions.py

```diff
@@ -65,68 +38,82 @@ def test_equality(some_paramspecs):
+        ser_desc = desc.serialize()
 
-    new_desc = RunDescriber.deserialize(ser_desc)
+        new_desc = RunDescriber.deserialize(ser_desc)
 
-    assert isinstance(new_desc, RunDescriber)
-    assert desc == new_desc
+        assert isinstance(new_desc, RunDescriber)
+        assert desc == new_desc
 
 
-def test_yaml_creation_and_loading(some_paramspecs):
+def test_yaml_creation_and_loading(some_interdeps):
+
     yaml = YAML()
-    for group in some_paramspecs.values():
-        paramspecs = group.values()
-        idp = InterDependencies(*paramspecs)
-        desc = RunDescriber(interdeps=idp)
 
-        yaml_str = desc.to_yaml()
+    for idps in some_interdeps:
+        desc = RunDescriber(interdeps=idps)
+
+        yaml_str = serial.make_yaml_for_storage(desc)
         assert isinstance(yaml_str, str)
         ydict = dict(yaml.load(yaml_str))
-        assert list(ydict.keys()) == ['interdependencies']
+        assert list(ydict.keys()) == ['version', 'interdependencies']
+        assert ydict['version'] == serial.STORAGE_VERSION
 
-        new_desc = RunDescriber.from_yaml(yaml_str)
+        new_desc = serial.read_yaml_to_current(yaml_str)
         assert new_desc == desc
 
 
-def test_jsonization_as_old(some_paramspecs):
+def test_default_jsonization_as_v0(some_interdeps):
     """
-    Test that a RunDescriber always json-ifies itself as an old style
-    RunDescriber, even when given new style interdeps
+    Test that a RunDescriber is json-dumped as version 0
     """
-
-    idps_old = InterDependencies(*some_paramspecs[2].values())
-    idps_new = old_to_new(idps_old)
+    idps_new = some_interdeps[0]
+    idps_old = new_to_old(idps_new)
 
     new_desc = RunDescriber(idps_new)
-    old_desc = RunDescriber(idps_old)
+    old_json = json.dumps({'version': 0,
+                           'interdependencies': idps_old.serialize()})
 
-    assert new_desc.to_json() == old_desc.to_json()
+    assert serial.make_json_for_storage(new_desc) == old_json
 
 
-def test_serialization_as_old(some_paramspecs):
+def test_default_serialization_as_v0(some_interdeps):
     """
     Test that a RunDescriber always serializes itself as an old style
     RunDescriber, even when given new style interdeps
     """
 
-    idps_old = InterDependencies(*some_paramspecs[2].values())
-    idps_new = old_to_new(idps_old)
+    idps_new = some_interdeps[0]
+    idps_old = new_to_old(idps_new)
 
     new_desc = RunDescriber(idps_new)
-    old_desc = RunDescriber(idps_old)
+    old_desc = {'version': 0, 'interdependencies': idps_old.serialize()}
+
+    assert serial.serialize_to_storage(new_desc) == old_desc
+
+
+def test_serialization_version_1(some_interdeps):
+    """
+    Test the serialization of a RunDescriber version 1 object
+    """
+    for idps in some_interdeps:
+        desc = RunDescriber(idps)
 
-    assert new_desc.serialize() == old_desc.serialize()
+        ser = desc.serialize()
+        assert ser['version'] == 1
```

Comment:
@astafan8 Given our discussion should this not serialize to version 0 still

Reply:
no, because `RunDescriber` here is of `version` `1`. So when calling deserialize method of `RunDescriber` of `version` `1` you get a dict with expected structure and it also has a field called `'version'` saying `1`. And i will rename the test for clarity.